From 74510049387ac9966c682ad8e6bc9eb7675ddbb5 Mon Sep 17 00:00:00 2001
From: James Christopher Adduono <jc@adduono.com>
Date: Wed, 6 Sep 2017 16:11:41 -0400
Subject: [PATCH 28/30] export is_cma_pageblock

---
 mm/page_alloc.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/page_alloc.c b/mm/page_alloc.c
index 3b767bb5e2f..6d8512875e9 100644
--- a/mm/page_alloc.c
+++ b/mm/page_alloc.c
@@ -793,6 +793,7 @@ bool is_cma_pageblock(struct page *page)
 {
 	return get_pageblock_migratetype(page) == MIGRATE_CMA;
 }
+EXPORT_SYMBOL(is_cma_pageblock);
 
 /* Free whole pageblock and set it's migration type to MIGRATE_CMA. */
 void __init init_cma_reserved_pageblock(struct page *page)
-- 
2.25.1

