From c996743b33f4b4b5dad3114724e1aa889b9fe4e7 Mon Sep 17 00:00:00 2001
From: kimocoder <christian@aircrack-ng.org>
Date: Wed, 12 Jan 2022 22:21:33 +0100
Subject: [PATCH] Makefile.lib: Stop calling size_append

---
 scripts/Makefile.lib | 13 ++++---------
 1 file changed, 4 insertions(+), 9 deletions(-)

diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index dcacdc946013..98edf735eb73 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -354,21 +354,18 @@ printf "%08x\n" $$dec_size |						\
 
 quiet_cmd_bzip2 = BZIP2   $@
 cmd_bzip2 = (cat $(filter-out FORCE,$^) | \
-	bzip2 -9 && $(call size_append, $(filter-out FORCE,$^))) > $@ || \
-	(rm -f $@ ; false)
+	bzip2 -9 || (rm -f $@ ; false)
 
 # Lzma
 # ---------------------------------------------------------------------------
 
 quiet_cmd_lzma = LZMA    $@
 cmd_lzma = (cat $(filter-out FORCE,$^) | \
-	lzma -9 && $(call size_append, $(filter-out FORCE,$^))) > $@ || \
-	(rm -f $@ ; false)
+	lzma -9 || (rm -f $@ ; false)
 
 quiet_cmd_lzo = LZO     $@
 cmd_lzo = (cat $(filter-out FORCE,$^) | \
-	lzop -9 && $(call size_append, $(filter-out FORCE,$^))) > $@ || \
-	(rm -f $@ ; false)
+	lzop -9 || (rm -f $@ ; false)
 
 quiet_cmd_lz4 = LZ4     $@
 cmd_lz4 = (cat $(filter-out FORCE,$^) | \
@@ -419,9 +416,7 @@ quiet_cmd_uimage = UIMAGE  $(UIMAGE_OUT)
 # decompression mode. A BCJ filter isn't used either.
 quiet_cmd_xzkern = XZKERN  $@
 cmd_xzkern = (cat $(filter-out FORCE,$^) | \
-	sh $(srctree)/scripts/xz_wrap.sh && \
-	$(call size_append, $(filter-out FORCE,$^))) > $@ || \
-	(rm -f $@ ; false)
+	sh $(srctree)/scripts/xz_wrap.sh || (rm -f $@ ; false)
 
 quiet_cmd_xzmisc = XZMISC  $@
 cmd_xzmisc = (cat $(filter-out FORCE,$^) | \
